ogmios/__init__.py

```diff
@@ -65,10 +65,22 @@ def get_template_source(self):
 
         for engine in engine_list:
             for loader in engine.engine.template_loaders:
-                try:
-                    return loader.load_template_source(self.filename)[0]
-                except TemplateDoesNotExist:
-                    pass
+                # Check for the cached template loader
+                if hasattr(loader, 'get_template_sources'):
```

Comment:
Yes, I read the comment. If you want to special case the cached template loader why don't you check for the cached template loader?